### PR TITLE
docs: Add Dynamic signer

### PR DESCRIPTION
**Pull request by matthew1809**
Adds Dynamic signer guide

### CODE REVIEW

**Comment by tanay1337** [comment]
@germartinez associated PR: https://github.com/5afe/safe-signers/pull/1

File: pages/sdk/signers/dynamic.mdx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+import { Tabs, Steps, Callout } from 'nextra/components'
+
+# Dynamic Signer
+
+In this guide, you will learn how to create a [Dynamic](https://dynamic.xyz) signer that can be added as a Safe owner and used to initialize any of the kits from the Safe\{Core\} SDK.
+
+Check out the [Safe Signers demo app](https://github.com/5afe/safe-signers) on GitHub to follow along this guide with the completed code.
+
+Note that this guide will focus on supporting `Ethereum`/`EVM` wallets. Simply enable, import and add [more connectors](https://docs.dynamic.xyz/chains/enabling-chains#enabling-a-chain-network) as needed.
```
Comment (tanay1337) [inline pages/sdk/signers/dynamic.mdx:11]:
```suggestion
Note that this guide will focus on supporting Ethereum and EVM-compatible wallets. You can enable, import and add [more connectors](https://docs.dynamic.xyz/chains/enabling-chains#enabling-a-chain-network) as needed.
```

File: pages/sdk/signers/dynamic.mdx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+## Prerequisites
+
+- A [Dynamic account](https://app.dynamic.xyz) and an Environment ID.
```
Comment (tanay1337) [inline pages/sdk/signers/dynamic.mdx:15]:
```suggestion
- A [Dynamic account](https://app.dynamic.xyz) and an environment ID.
```

File: pages/sdk/signers/dynamic.mdx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    <Tabs.Tab>
+      You can instantiate the provider using `viem` and the following imports:
+
+      ```typescript
+      useEffect(() => {
+        const init = async () => {
+          if (isLoggedIn && primaryWallet && isEthereumWallet(primaryWallet)) {
+            const provider = await primaryWallet.getWalletClient()
+
+            const signerAddress = primaryWallet.address
```
Comment (germartinez) [inline pages/sdk/signers/dynamic.mdx:126]:
```suggestion
            const signer = primaryWallet.address
```

File: pages/sdk/signers/dynamic.mdx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+  ### Get the provider and signer
+
+  Once the user is logged in through the DynamicWidget, you can get the `provider` and `signerAddress`, which is the externally owned account of the user that was derived from its credentials.
+  
+  To do that there is a `useEffect()` that executes when any of the `isLoggedIn` and `primaryWallet` variables have their values updated. Once they are all true, you have access to the user's connected signer. Note that we are also checking that the wallet is an Ethereum wallet using the `isEthereumWallet()` function.
+
+{/* <!-- vale off --> */}
+
+  <Tabs items={['viem']}>
```
Comment (germartinez) [inline pages/sdk/signers/dynamic.mdx:116]: If `viem` is not used, the `Tabs` (including the import) should be removed.

File: pages/sdk/signers/dynamic.mdx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+
+  {/* <!-- vale on --> */}
+  
+  With the `provider` and `signerAddress` you are ready to instantiate any of the kits from the Safe\{Core\} SDK and set up or use this signer as a Safe owner.
+
+</Steps>
+
+## Recap and further reading
+
+After following this guide, you are able to create a Safe signer using Dynamic and get the `provider` and `signerAddress` required to initialize the kits from the Safe\{Core\} SDK.
```
Comment (germartinez) [inline pages/sdk/signers/dynamic.mdx:143]:
```suggestion
After following this guide, you are able to create a Safe signer using Dynamic and get the `provider` and `signer` required to initialize the kits from the Safe\{Core\} SDK.
```

**Review by germartinez — CHANGES_REQUESTED** [review]

**Comment by germartinez** [comment]
@matthew1809 thanks for the great PR!
We just added some small suggestions before this is merged.

PD: I'm also replacing `signerAddress` with `signer` in all the other signers guides, not just this one ;)

File: pages/sdk/signers/dynamic.mdx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+  {/* <!-- vale off --> */}
+
+  ```typescript
+  import { DynamicContextProvider, useDynamicContext, useIsLoggedIn } from '@dynamic-labs/sdk-react-core'
+  import { EthereumWalletConnectors, isEthereumWallet } from '@dynamic-labs/ethereum'
+  ```
+
+  {/* <!-- vale on --> */}
+
```
[the remark this answers is not in the document]
Comment (matthew1809) [reply pages/sdk/signers/dynamic.mdx:57]: Thanks, fixed

File: pages/sdk/signers/dynamic.mdx (hunk 4 above)
Comment (matthew1809) [reply pages/sdk/signers/dynamic.mdx]: Done

**Review by germartinez — APPROVED** [review]

**Pull request merged**